hadoop-ozone/ozone-recon/src/main/java/org/apache/hadoop/ozone/recon/spi/impl/ContainerDBServiceProviderImpl.java

```diff
@@ -181,6 +182,12 @@ public Integer getCountForForContainerKeyPrefix(
       Long containerID = keyValue.getKey().getContainerId();
       Integer numberOfKeys = keyValue.getValue();
 
+      // break the loop if limit has been reached
+      // and one more new entity needs to be added to the containers map
+      if (containers.size() == limit && !containers.containsKey(containerID)) {
```

Comment:
Not understood the reason for the 2nd condition.
And this method can be called with limit -1, Can you update java doc with description, what happens in that case.

Reply:
Without the second condition, the last container ID will have incorrect number of keys in its containerMetadata. Even when containers limit is reached, the next iteration could contain the same container ID with a different key prefix. To handle this situation, we should not break the iterator until the second condition is also met.

I don't think we need to add any check for any other negative values since none of them match the condition `containers.size() == limit` at any time.

Reply:
Got it..